diag_manager/fms_diag_time_reduction.F90

```diff
@@ -0,0 +1,218 @@
+
+   !!These parametes are the possible kinds of time reduction operations.
+   !!Note that sometimes one kind inplies another.
+   !!TODO: should they be put in diag_data ?
+   !!TODO:
+   !!TODO: time_diurnal "not really" same kind as others, so remove?
+   INTEGER, PARAMETER :: time_none    = 0 !< There is no reduction method
+   INTEGER, PARAMETER :: time_average = 1 !< The reduction method is avera
+   INTEGER, PARAMETER :: time_rms     = 2 !< The reduction method is rms
+   INTEGER, PARAMETER :: time_max     = 3 !< The reduction method is max
+   INTEGER, PARAMETER :: time_min     = 4 !< The reduction method is min
+   INTEGER, PARAMETER :: time_sum     = 5 !< The reudction method is sum
+   INTEGER, PARAMETER :: time_diurnal = 6 !< The reduction method is diurnal
+   INTEGER, PARAMETER :: time_power   = 7 !< The reduction method is power
+
+!> @brief Class time_reduction_type has an encapsulation of the "Fortran enum" time
+!! reduction integer parameters, plus an encapsulation of the groupings of
+!! the time reduction types. It is inteded to provide some of the functionality
+!! that was coded in the legacy function diag_data.F90:init_output_fields.
+!! The functionality in the end is used by send_data in (EFFICIENT) do loops calling
+!! the weighting or math functions to update buffers.
+!!  the The integer parameters above are the legal time_reduction_types,
+!! but they are not necessarily mutually exclusive in some contexts.
+!!
+!> @addtogroup fms_diag_time_reduction_mod
+   TYPE time_reduction_type
+      integer , private :: the_type !< The time reduction type; integer as per diag_data_mod entries.
+      logical , private :: time_averaging !< Set true iff time_average, time_rms, time_power or time_diurnal is true
+      logical , private :: time_ops !< Set true iff time_min, time_max, time_rms or time_average is true.
+   CONTAINS
+      procedure, public :: do_time_averaging => do_time_averaging_imp
+      procedure, public :: has_time_ops => has_time_ops_imp
+      procedure, public :: is_time_none => is_time_none_imp
+      procedure, public :: is_time_average => is_time_average_imp
+      procedure, public :: is_time_rms => is_time_rms_imp
+      procedure, public :: is_time_max => is_time_max_imp
+      procedure, public :: is_time_min => is_time_min_imp
+      procedure, public :: is_time_sum => is_time_sum_imp
+      procedure, public :: is_time_diurnal => is_time_diurnal_imp
+      procedure, public :: is_time_power => is_time_power_imp
+      procedure, public :: initialize
+   END TYPE time_reduction_type
+
+!> @brief This interface is for the class constructor.
+!> @addtogroup fms_diag_time_reduction_mod
+   interface  time_reduction_type
+      procedure  :: time_reduction_type_constructor
+   end interface time_reduction_type
+
+CONTAINS
+
+   !> @brief The class contructors. Just allocates the class and calls an initializer
+   function time_reduction_type_constructor(dt, out_frequency) result(time_redux)
```

Comment:
Needs a doxygen comment  describing the result (`time_redux`) variable

Reply:
done